extensions/mssql/src/objectExplorerNodeProvider/connection.ts

```diff
@@ -188,15 +189,23 @@ export class Connection {
 		return this.connectionInfo.options[constants.groupIdName];
 	}
 
-	public isMatch(connectionInfo: sqlops.ConnectionInfo): boolean {
+	public async isMatch(connectionInfo: sqlops.ConnectionInfo): Promise<boolean> {
 		if (!connectionInfo) {
 			return false;
 		}
-		let otherConnection = new Connection(connectionInfo);
-		return otherConnection.groupId === this.groupId
-			&& otherConnection.host === this.host
-			&& otherConnection.knoxport === this.knoxport
-			&& otherConnection.user === this.user;
+		let profile = connectionInfo as sqlops.IConnectionProfile;
+		if (profile) {
+			let result: IEndpoint = await utils.getClusterEndpoint(profile.id, constants.hadoopKnoxEndpointName);
+			if (result === undefined || !result.ipAddress || !result.port) {
+				return false;
+			}
+			return connectionInfo.options.groupId === this.groupId
+				&& result.ipAddress === this.host
+				&& String(result.port).startsWith(this.knoxport)
+				&& String(result.port).endsWith(this.knoxport);
+				// TODO: enable the user check when the unified user is used
```

Comment:
Is this TODO relevant? Do you mean when we no longer just use `root` in this case?